src/redux/missions/missions.js

```diff
@@ -0,0 +1,41 @@
+const SUCCESS_MISSION = 'SUCCESS_MISSION';
+const JOIN_MISSION = 'JOIN_MISSION';
+
+const initialState = [];
+
+export const successMission = (payload) => ({
+  type: SUCCESS_MISSION,
+  payload,
+});
+
+export const joinMission = (payload) => ({
+  type: JOIN_MISSION,
+  payload,
+});
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SUCCESS_MISSION: {
+      if (state.length === 0) {
+        const newState = action.payload.map((element) => ({
+          mission_id: element.mission_id,
+          mission_name: element.mission_name,
+          description: element.description,
+        }));
+        return newState;
+      }
+      return state;
+    }
+    case JOIN_MISSION: {
+      const newState = state.map((mission) => {
+        if (mission.mission_id !== action.payload.id) return mission;
+        return { ...mission, member: !mission.member, mission_button: !mission.mission_button };
+      });
+      return newState;
+    }
+    default:
+      return state;
+  }
+};
+
+export default reducer;
```

Comment:
- [ ] Great implementation here, however, let us modify our structure in this way, the below structure will help us to split out the different actions in different folders not in one folder as you did in the `src/redux/missions/missions.js`

Create two folders in the redux folder: 

- Actions
    - actiontypes.js
       - export const SUCCESS_MISSION = 'SUCCESS_MISSION';
       - export const JOIN_MISSION = 'JOIN_MISSION';
    - actioncreator.js
       - export const successMission = (payload) => ({
       type: SUCCESS_MISSION,
       payload,
       });
- Reducers to keep the initialState and our cases
    - missionReducer.js
    - rocketReducers.js
    
The above structure will help us to make a clear implementation of our tasks as well as to create many actions to be implemented as long as we have a big project.